src/main/java/bio/terra/janitor/app/controller/UnauthenticatedApiController.java

```diff
@@ -2,14 +2,43 @@
 
 import bio.terra.generated.controller.UnauthenticatedApi;
 import bio.terra.generated.model.SystemStatus;
+import bio.terra.generated.model.SystemStatusSystems;
+import bio.terra.janitor.app.configuration.JanitorJdbcConfiguration;
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Controller;
 
 @Controller
 public class UnauthenticatedApiController implements UnauthenticatedApi {
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Autowired
+  UnauthenticatedApiController(JanitorJdbcConfiguration jdbcConfiguration) {
+    this.jdbcTemplate = new NamedParameterJdbcTemplate(jdbcConfiguration.getDataSource());
+  }
+
   @Override
   public ResponseEntity<SystemStatus> serviceStatus() {
-    return new ResponseEntity<>(new SystemStatus().ok(true), HttpStatus.OK);
+    if (jdbcTemplate.getJdbcTemplate().execute(this::isConnectionValid)) {
```

Comment:
wdyt about inlining `isConnectionValid`? don't think we're getting much readability benefit from having a named function.

`.execute(connection - > connection.isValid(0)) {...`

Reply:
done